### PR TITLE
Add `*.js` export variants for compat files

### PR DESCRIPTION
**Pull request by philipp-spiess**
In v3 it was possible to import `*.js` variants of compat files, e.g. `tailwindcss/plugin.js` in addition to `tailwindcss/plugin`. We need to properly map the exports to support this.

### CODE REVIEW

**Comment by thecrypticace** [comment]
Are there projects / plugins / whatever actually doing this?

**Comment by philipp-spiess** [comment]
@thecrypticace loads: https://sourcegraph.com/search?q=context:global+%22from+%27tailwindcss/plugin%22&patternType=keyword&sm=0

**Comment by thecrypticace** [comment]
About 15–16% wow. alright 👍

File: packages/tailwindcss/package.json [hunk 1]
```diff
@@ -27,14 +27,26 @@
...
+      "import": "./src/compat/default-theme.ts"
+    },
     "./plugin": {
       "require": "./src/plugin.cts",
       "import": "./src/plugin.ts"
     },
+    "./plugin.js": {
+      "require": "./src/plugin.cts",
+      "import": "./src/plugin.ts"
+    },
```
Comment (thecrypticace) [inline packages/tailwindcss/package.json:49]: These need to be updated in the publishConfig too

**Review by thecrypticace — DISMISSED** [review]
small change

File: packages/tailwindcss/package.json (hunk 1 above)
Comment (philipp-spiess) [reply packages/tailwindcss/package.json:49]: ah of course 🫠

**Review by thecrypticace — APPROVED** [review]

**Pull request merged**